manifests/config.pp

```diff
@@ -68,7 +68,9 @@
     'database'  => $foreman::db_database,
     'username'  => $foreman::db_username,
     'password'  => $foreman::db_password,
-    'db_pool'   => max($foreman::db_pool, $foreman::foreman_service_puma_threads_max),
+    # Set the pool size to at least the amount of puma threads + 4 threads that are spawned automatically by the process.
+    # db_pool is optional, and undef means "use default" and the second part of the max statement will be set
+    'db_pool'   => max(pick($foreman::db_pool, 0), $foreman::foreman_service_puma_threads_max + 4),
```

Comment:
I'd keep this simple so you still provide the user with full control. For example if the `+ 4` part does give problems you still provide a way to override it.
```suggestion
    'db_pool'   => pick($foreman::db_pool, $foreman::foreman_service_puma_threads_max + 4),
```